dm/syncer/data_validator.go

```diff
@@ -355,16 +366,21 @@ func (v *DataValidator) errorProcessRoutine() {
 func (v *DataValidator) waitSyncerSynced(currLoc binlog.Location) error {
 	syncLoc := v.syncer.getFlushedGlobalPoint()
 	cmp := binlog.CompareLocation(currLoc, syncLoc, v.cfg.EnableGTID)
-	if cmp <= 0 {
+	switch {
+	case cmp < 0:
 		return nil
+	case cmp == 0:
+		v.reachedSyncer.Store(true)
```

Comment:
I suddenly forget why reachedSyncer is true when currLoc >= syncLoc 😂

Reply:
when `reachedSyncer=true`, validator will start to mark rows as error rows, before that we don't, to reduce chance of false-negative